src/argilla/client/feedback/dataset/base.py

```diff
@@ -306,14 +313,27 @@ def unify_responses(
     def prepare_for_training(
         self,
         framework: Union[Framework, str],
-        task_mapping: TrainingTaskMappingForTextClassification,
+        task: TrainingTaskTypes,
         train_size: Optional[float] = 1,
         test_size: Optional[float] = None,
         seed: Optional[int] = None,
         lang: Optional[str] = None,
         fetch_records: Optional[bool] = None,
     ):
-        # TODO(davidberenstein1957): add missing docstrings and type annotations
+        """
+        Prepares the dataset for training for a specific training framework and NLP task by splitting the dataset into train and test sets.
+
+        Args:
+            framework: the framework to use for training. Currently supported frameworks are: `transformers`, `peft`,
+                `setfit`, `spacy`, `spacy-transformers`, `span_marker`, `spark-nlp`, `openai`, `trl`.
+            task: the NLP task to use for training. Currently supported tasks are: `TrainingTaskForTextClassification`,
+                `TrainingTaskForSFT`, `TrainingTaskForRM`, `TrainingTaskForPPO`, `TrainingTaskForDPO`.
+            train_size: the size of the train set. If `None`, the whole dataset will be used for training.
+            test_size: the size of the test set. If `None`, the whole dataset will be used for testing.
+            seed: the seed to use for splitting the dataset into train and test sets.
+            lang: the spaCy language to use for training. If `None`, the language of the dataset will be used.
+            fetch_records: whether to fetch the records from Argilla or use the local records instead. If `None`, use local.
+        """
```

Comment:
`fetch_records` will be deprecated from 1.15.0 onwards, so maybe we can just remove this and iterate over `dataset.records`, since it will fetch those from Argilla if not there, or just loop over those

Reply:
```suggestion
        """
```

Reply:
@alvarobartt I agree, this aligns with the mention I had regarding the differentiation between local and remote datasets.

Reply:
I'll delete it from the ArgillaTrainer. Should I also remove it from `prepare_for_training`? I think that might break backwards compatibility.

Reply:
2b576dfa6be51eaa106c96623a56f37fbea9de06

Reply:
As discussed privately in Slack, we'll keep this for a separate deprecation PR.